### ISSUE TITLE
[Request] Node suggestions

### ISSUE THREAD

**Issue by WindySIn** [zenoverflow/omnichain#12]
I've been playing with this for the last week and I gotta say I'm really pleased with how this all works. If I had a gripe it would maybe be that the backwards flowing nature of message requests pulling data is a little confusing until you get used to it. Anyway, there's a few nodes I think would be useful:

- Something like a Java switch statement. The existing regex node does the job, but if you want to test several possible regex options then it gets clunky. For example, I've had good success telling the LLM to evaluate a prompt and output a single word from a list of options in order to drive a branching narrative / menu. Having a single node to then direct to the correct branches would be useful, and less clunky than having to nest several regex nodes.
- Random selection from array. Alternatively, an RNG node that can be plugged into the pick from array nodes.
- Edit JSON string by key, and validate JSON string. The ability to get info from JSON is nice, but being able to manipulate it and sanitise outputs from the LLM would allow a JSON data structure to back the LLM's inference.

Anyway, hope some of these are useful and/or practical to implement. Love the stuff you've got so far. By the by, in case anyone's curious this works fine with Koboldcpp using the OAI chat completion interface.

**Comment by zenoverflow** [comment]
Those are all great ideas! I'll look into the best way to implement each and update this issue when I have something working.

About the backwards flow thing - that's just how data-flow graphs work (same as ComfyUI). Control-flow graphs are reversed, but a pure control-flow graph has complexity issues when combining different data pipelines. I went through a few variants before deciding on this hybrid architecture as the best of both worlds.

In regards to Kobold - I'm planning on making dedicated nodes for both that and pure llamacpp, although yeah you can use the OpenAI nodes too afaik, they just don't have the custom samplers.

**Comment by WindySIn** [comment]
The comment about the backwards flow is in no way a complaint about the design. I use ComfyUI quite a lot so I get it. At most, I would suggest that it might be handy to have maybe some visual highlight for the whole chain if you click on a response node, for example. Sometimes in a busy workflow it gets hard to follow the chain backwards.

**Comment by zenoverflow** [comment]
I've added a random number generator node, random array picker nodes, a json editor node, and a json validator node. I'll look into the switch node tomorrow. For now, you can do a `git pull` and `npm install` to update your instance.

**Comment by WindySIn** [comment]
That looks pretty good. I've got a few busy days ahead but I'll have a play with the new nodes soon as I can. Thanks for being so responsive. I think the chain highlight idea is very much optional. Truth be told, I don't know how much utility it'll really have.

**Comment by av** [comment]
Dify has a "classifier" node that can use an LLM to categorise the incoming message into one of the predefined classes.

I think it can be arranged as a custom node with two inputs:
```
// Text to classify
text: string

// Plain text descriptions of the variants to choose from
classes: string[]
```
The node can output the matched index, to feed index triggers from above screenshots

**Comment by zenoverflow** [comment]
> Dify has a "classifier" node that can use an LLM to categorise the incoming message into one of the predefined classes.
> 
> I think it can be arranged as a custom node with two inputs:
> 
> ```
> // Text to classify
> text: string
> 
> // Plain text descriptions of the variants to choose from
> classes: string[]
> ```
> 
> The node can output the matched index, to feed index triggers from above screenshots

You can (and are supposed) to do this with the template builder, the text joiner, any LLM node, and the regex extractor.
